src/pal/src/misc/time.cpp

```diff
@@ -273,13 +256,26 @@ QueryPerformanceFrequency(
     }
     else
     {
-        lpFrequency->QuadPart = (LONGLONG)tccSecondsToNanoSeconds * ((LONGLONG)s_TimebaseInfo.denom / (LONGLONG)s_TimebaseInfo.numer);
+        // (numer / denom) gives you the nanoseconds per tick, so the below code
+        // computes the number of ticks per second. We explicitly do the multiplication
+        // first in order to help minimize the error that is produced by integer division.
+
+        lpFrequency->QuadPart = ((LONGLONG)(tccSecondsToNanoSeconds) * (LONGLONG)(s_TimebaseInfo.denom)) / (LONGLONG)(s_TimebaseInfo.numer);
```

Comment:
This is not correct, the original version was. The result of QueryPerformanceCount divided by the QueryPerformanceFrequency is supposed to give you time in seconds. The QueryPerformanceCount always returns time in nanoseconds for the monotonic clock, so the QueryPerformanceFrequency should return number of nanoseconds per second.

Reply:
I was not talking about Mac or the order of division and multiplication. Those are good. But I was referring to the monotonic clock. Before your change, the QueryPerformanceFrequency was returning a fixed value number of nanoseconds per second. Now it returns a value computed from the `s_Timespec.tv_sec` and `s_Timespec.tv_nsec`, a number of ticks per second (a precision of the monotonic clock). But the QueryPerformanceCounter returns number of nanoseconds elapsed, not a number of ticks (since the `clock_gettime` always returns seconds and nanoseconds and we just convert the seconds to nanoseconds, add the nanoseconds part and return that as the result.

Reply:
> I was not talking about Mac or the order of division and multiplication.

Ah, the highlighted line is the Mac one, hence the confusion.

>  But the QueryPerformanceCounter returns number of nanoseconds elapsed

This is not true. `QPC` returns a count that has no direct correlation to any specific measurement of time and must be scaled by `QPF` (which dictates the number of counts per second) to get an actual measurement of time (also see https://docs.microsoft.com/en-us/windows/desktop/SysInfo/acquiring-high-resolution-time-stamps).

POSIX systems using `clock_gettime` will ultimately get a result that has a peak resolution in terms of `nanoseconds`, but will actually be some multiple of `clock_getres`. Having the actual resolution then, ultimately, will just truncate any non-significant digits from the `clock_gettime` result. On desktop Linux, `clock_getres(CLOCK_MONOTONIC)` is currently hardcoded to `tv_sec = 0; tv_nsec = 1`, so this will have no real impact other than making the code more correct/portable.

Reply:
Basically, the previous logic was assuming the hardcoded value of and so it returned `tccSecondsToNanoSeconds` (`1e9`).

The new logic takes the actual resolution into account. So, for example, if the resolution was `tv_sec = 0; tv_nsec = 2`, it would now return `500,000,000` (which is correctly 1 tick every 2ns). Likewise if the underlying resolution was `tv_sec = 0; tv_nsec = 6,000,000`, we would get `166` (which is 1 tick per `~0.006` seconds)

Reply:
> This is not true. `QPC` returns a count that has no direct correlation to any specific measurement of time and must be scaled by `QPF` (which dictates the number of counts per second) to get an actual measurement of time.

In the current implementation with monotonic clock, QueryPerformanceCounter does always return number of nanoseconds, no matter what the `clock_getres` returns. We call the `clock_gettime` and then compute the returned value as 
```c++
((LONGLONG)(ts.tv_sec) * (LONGLONG)(tccSecondsToNanoSeconds)) + (LONGLONG)(ts.tv_nsec);
```

Reply:
Oh, I see. Right.

Since `clock_gettime` returns it in terms of ns, rather than in terms of an arbitrary `count`, you would have to divide this by the resolution to get a valid `count`. You then have to decide if you want something that is "closer" to what the other systems deal with or if you just want to return everything in terms of nsec...

I'll revert this back to `tccSecondsToNanoSeconds` since that will allow the highest throughput and will minimize error propagated up.